src/dune_rules/alias_rec.ml

```diff
@@ -1,23 +1,122 @@
 open Import
 
-include Action_builder.Alias_rec (struct
+module Alias_status = struct
+  module T = struct
+    type t =
+      | Defined
+      | Not_defined
+
+    let empty : t = Not_defined
+
+    let combine : t -> t -> t =
+     fun x y ->
+      match (x, y) with
+      | _, Defined | Defined, _ -> Defined
+      | Not_defined, Not_defined -> Not_defined
+  end
+
+  include T
+  include Monoid.Make (T)
+end
+
+module Source_tree_only = struct
+  let dep_on_alias_if_exists alias =
+    let open Action_builder.O in
+    let* definition = Action_builder.of_memo (Load_rules.alias_exists alias) in
+    match definition with
+    | false -> Action_builder.return Alias_status.Not_defined
```

Comment:
~If I understand correctly, this allows to depend on aliases that don't exist? I wonder if this should be an error.~ Edit: never mind, i see we accumulate aliases statuses and error out at the end if nothing has been found.